pdns/lua-recursor4.cc

```diff
@@ -516,8 +554,11 @@ bool RecursorLua4::genhook(luacall_t& func, const ComboAddress& remote,const Com
   dq->ednsOptions = ednsOpts;
   dq->isTcp = isTcp;
   dq->policyTags = policyTags;
+  dq->appliedPolicy = appliedPolicy;
+  dq->wantsRPZ = wantsRPZ;
   bool handled=func(dq);
   if(variable) *variable |= dq->variable; // could still be set to indicate this *name* is variable, even if not 'handled'
+  *wantsRPZ = dq->wantsRPZ; // Even if we did not handle the query, RPZ could be disabled
```

Comment:
That shouldn't be an issue, at that point, all RPZ related code is run. but NULL might be bad though